Slack.Automation/Promact.Core.Repository/SlackRepository/SlackRepository.cs

```diff
@@ -38,7 +42,111 @@ public SlackRepository(ILeaveRequestRepository leaveRepository, IOauthCallsRepos
             _userManager = userManager;
             _slackUserRepository = slackUserRepository;
         }
+        #endregion
 
+        #region Public Methods
+        /// <summary>
+        /// Method to get leave Updated from slack button response
+        /// </summary>
+        /// <param name="leaveId"></param>
+        /// <param name="status"></param>
+        /// <returns>replyText</returns>
+        public void UpdateLeave(SlashChatUpdateResponse leaveResponse)
+        {
+            // method to get leave by its id
+            var leave = _leaveRepository.LeaveById(leaveResponse.CallbackId);
+            // only pending status can be modified
+            if (leave.Status == Condition.Pending)
+            {
+                if (leaveResponse.Actions.Value == _stringConstant.Approved)
+                {
+                    leave.Status = Condition.Approved;
+                }
+                else
+                {
+                    leave.Status = Condition.Rejected;
+                }
+            }
+            _leaveRepository.UpdateLeave(leave);
+
+            replyText = string.Format("You had {0} Leave for {1} From {2} To {3} for Reason {4} will re-join by {5}",
+                            leave.Status,
+                            leaveResponse.User.Name,
+                            leave.FromDate.ToShortDateString(),
+                            leave.EndDate.Value.ToShortDateString(),
+                            leave.Reason,
+                            leave.RejoinDate.Value.ToShortDateString());
+            _client.UpdateMessageAsync(leaveResponse, replyText);
```

Comment:
this method is async and should be awaited. the parent method is not marked as async too